xwiki-platform-core/xwiki-platform-office/xwiki-platform-office-importer/src/main/java/org/xwiki/officeimporter/internal/filter/HeadingInOrderedListFilter.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.officeimporter.internal.filter;
+
+import java.util.Map;
+import java.util.List;
+import java.util.ArrayList;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.xml.html.filter.AbstractHTMLFilter;
+
+
+/**
+ * Sometimes the heading becomes wrapped in an ordered list structure such as this during import:
+ * 1.
+ * 
+ * 11. 
+ * 111. 
+ * 1111. 
+ * 11111. (((
+ * ===== Heading text =====
+ * )))
+ *
+ * This filter removes these structures.
```

Comment:
Would be good to indicate what is the result after the filter is applied, otherwise it's not very clear what you mean by "these structures".